### PR TITLE
Delighted access and erasure

### LINKED ISSUE #1880: Delighted

### PR DESCRIPTION
**Pull request by Kelsey-Ethyca**
Closes #1880

### Code Changes

* [x] Access and erasure support for Delighted
  * [x] person (read and erase)
  * [x] survey_response (read)
* [x] Can initiate traversal using email identity

### Steps to Confirm

* [x] access test with identity email
* [x] erasure test with generated erasure data

### Pre-Review Checklist

* [x] Postman collections are up-to-date
* [x] Test credentials have been added to 1password
* [x] The new tests for this connector pass locally

### CODE REVIEW

File: tests/ops/integration_tests/saas/test_delighted_task.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
+import random
```
Comment (sanders41) [inline tests/ops/integration_tests/saas/test_delighted_task.py:1]:
```suggestion
import uuid
```

File: tests/ops/integration_tests/saas/test_delighted_task.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    db,
+    policy,
+    delighted_connection_config,
+    delighted_dataset_config,
+    delighted_identity_email,
+) -> None:
+    """Full access request based on the Delighted SaaS config"""
+
+    privacy_request = PrivacyRequest(
+        id=f"test_delighted_access_request_task_{random.randint(0, 1000)}"
```
Comment (sanders41) [inline tests/ops/integration_tests/saas/test_delighted_task.py:35]: I haven't reviewed the whole PR, this spot just caught my eye. We have had flaky tests from using the random int in places like this before. Random between 0 and 1000 seems unlikely to clash, but we have seen it happen more than I would expect. Switching to a `uuid` will keep this from happening.

````suggestion
        id=f"test_delighted_access_request_task_{uuid.uuid4()}"
````

Comment (Kelsey-Ethyca) [reply tests/ops/integration_tests/saas/test_delighted_task.py:35]: This is boiler plate code in all our connector tests. Let me create a PR to fix this in all our tests.

**Review by galvana — APPROVED** [review]
Looks good to me 👍

**Pull request merged**